gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/configuration/dictionary/DictionaryServiceImpl.java

```diff
@@ -295,15 +295,19 @@ public void delete(String id) {
                 throw new DictionaryNotFoundException(id);
             }
 
-            // Force un-deployment
-            if (dictionary.get().getType() == DictionaryType.MANUAL) {
-                undeploy(id);
+            if (dictionary.get().getType() == DictionaryType.DYNAMIC) {
+                this.stop(id);
             }
 
-            this.stop(id);
-
             dictionaryRepository.delete(id);
-        } catch (TechnicalException ex) {
+
+            Map<String, String> properties = new HashMap<>();
+            properties.put(Event.EventProperties.DICTIONARY_ID.getValue(), id);
+
+            // And create event
+            eventService.create(EventType.UNPUBLISH_DICTIONARY, mapper.writeValueAsString(dictionary), properties);
```

Comment:
for an UNPUBLISH event, the payload can be null
```suggestion
            eventService.create(EventType.UNPUBLISH_DICTIONARY, null, properties);
```

Reply:
Sûre ? In older undeploy method, we set the payload in unpublish event